fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java

```diff
@@ -523,24 +486,15 @@ public void sendConsensusElect(
     ConsensusElect consensusElect = new ConsensusElect(creation, objId, type, property, value);
 
     Log.d(TAG, PUBLISH_MESSAGE);
-    MessageGeneral msg = new MessageGeneral(mKeyManager.getMainKeyPair(), consensusElect, mGson);
+    MessageGeneral msg = new MessageGeneral(keyManager.getMainKeyPair(), consensusElect, gson);
 
     Disposable disposable =
-        mLAORepository
-            .sendPublish(channel, msg)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .timeout(5, TimeUnit.SECONDS)
+        networkManager
+            .getMessageSender()
+            .publish(channel, msg)
             .subscribe(
-                answer -> {
-                  if (answer instanceof Result) {
-                    Log.d(TAG, "created a consensus with messageId: " + msg.getMessageId());
-                  } else {
-                    Log.d(TAG, "failed to create a consensus");
-                  }
-                },
-                throwable ->
-                    Log.d(TAG, "timed out waiting for result on consensus/elect", throwable));
```

Comment:
here and elsewhere, I would keep this distinction.
Maybe we can have a utility function to deal with the logging+`toast` of timeouts vs. JSON-RPC errors.

Reply:
Added to the todo list